core/src/main/java/hudson/model/RunMap.java

```diff
@@ -228,9 +228,11 @@ protected R retrieve(File d) throws IOException {
                 }
                 return b;
             } catch (IOException e) {
-                LOGGER.log(Level.WARNING, "could not load " + d, e);
+                throw new IOException();
```

Comment:
There is no need to wrap it (the caller in `AbstractLazyLoadRunMap` includes the exact same information), but yes you do need to write this as

```java
throw e;
```

so as to not lose the original stack trace and detail message.